linera-execution/src/runtime.rs

```diff
@@ -112,13 +112,16 @@ struct ApplicationStatus {
     signer: Option<Owner>,
     /// The current execution outcome of the application.
     outcome: RawExecutionOutcome<Vec<u8>>,
+    /// The chain ID that created the application.
+    creator_chain_id: ChainId,
```

Comment:
Why do we need this here?

Reply:
We need it to get the `application_creator_chain_id`, because now the `ApplicationId` doesn't have the `creation` message anymore, which contained the creator `chain_id`. I thought it was better to just add it here than add it to the `ApplicationId`, but open to other suggestions 🤔